pkg/comp-functions/functions/vshnkeycloak/deploy.go

```diff
@@ -0,0 +1,373 @@
+
+	return res, nil
+}
+
+func newValues(ctx context.Context, svc *runtime.ServiceRuntime, comp *vshnv1.VSHNKeycloak, adminSecret string) (map[string]any, error) {
+	values := map[string]any{}
+
+	cd, err := svc.GetObservedComposedResourceConnectionDetails(comp.GetName() + pgInstanceNameSuffix)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := getResources(ctx, svc, comp)
+	if err != nil {
+		return nil, err
+	}
+
+	plan := comp.Spec.Parameters.Size.GetPlan(svc.Config.Data["defaultPlan"])
+	nodeSelector, err := utils.FetchNodeSelectorFromConfig(ctx, svc, plan, comp.Spec.Parameters.Scheduling.NodeSelector)
+	if err != nil {
+		return values, fmt.Errorf("cannot fetch nodeSelector from the composition config: %w", err)
+	}
+
+	fqdn := ""
+	strictHostname := false
+	if comp.Spec.Parameters.Service.FQDN != "" {
+		fqdn = comp.Spec.Parameters.Service.FQDN
+		strictHostname = true
+	}
+
+	extraEnv, err := toYaml([]map[string]any{
+		{
+			"name": "JAVA_OPTS",
+			"value": `-XX:+UseContainerSupport
+-XX:MaxRAMPercentage=50.0
+-Djava.net.preferIPv4Stack=true
+-Djava.awt.headless=true
+-Djgroups.dns.query={{ include "keycloak.fullname" . }}-headless`,
+		},
+		{
+			"name": "KEYCLOAK_ADMIN_PASSWORD",
+			"valueFrom": map[string]any{
+				"secretKeyRef": map[string]any{
+					"name": adminSecret,
+					"key":  "password",
+				},
+			},
+		},
+		{
+			"name":  "KEYCLOAK_ADMIN",
+			"value": "admin",
+		},
+		{
+			"name":  "KC_HOSTNAME",
+			"value": fqdn,
+		},
+		{
+			"name":  "KC_HOSTNAME_STRICT",
+			"value": strconv.FormatBool(strictHostname),
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	extraVolumes, err := toYaml([]map[string]any{
+		{
+			"name": "db-certs",
+			"secret": map[string]any{
+				"secretName":  pgSecretName,
+				"defaultMode": 0400,
+				"items": []map[string]any{
+					{
+						"key":  "tls.crt",
+						"path": "tls.crt",
+					},
+				},
+			},
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	extraVolumeMounts, err := toYaml([]map[string]any{
+		{
+			"name":      "db-certs",
+			"readOnly":  true,
+			"mountPath": "/opt/keycloak/db-certs",
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	startupProbe, err := toYaml(map[string]any{
+		"httpGet": map[string]string{
+			"path": `{{ tpl .Values.http.relativePath $ | trimSuffix "/" }}/health`,
+			"port": "http",
+		},
+		"intialDelaySeconds": 90,
+		"timeoutSeconds":     1,
+		"failureThreshold":   60,
+		"periodSeconds":      5,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	values = map[string]any{
+		"replicas": "1",
+		"image": map[string]any{
+			"tag": "23.0.2",
+		},
+		"resources": map[string]any{
+			"requests": map[string]any{
+				"memory": res.ReqMem,
+				"cpu":    res.ReqCPU,
+			},
+			"limits": map[string]any{
+				"memory": res.Mem,
+				"cpu":    res.CPU,
+			},
+		},
+		"nodeSelector":      nodeSelector,
+		"args":              []string{"start", "--http-enabled=true"},
```

Comment:
I don't think HTTP should be enabled for new setups. The keycloakx helm chart is somehow security broken. https://github.com/projectsyn/component-keycloak/issues/142

Reply:
TLS and all it's headaches will be part of another story. Here we're implementing the basic bootstrapping.

Reply:
Kubernetes itself doesn't validate the TLS certificate. So health checks and liveness probes work out of the box. But also here the keycloakx helm chart - with the sting handling - does make it more complicated than it should be https://github.com/projectsyn/component-keycloak/pull/182 as reference.